app/controllers/prison/visits_controller.rb

```diff
@@ -18,14 +18,15 @@ def process_visit
   # rubocop:disable Metrics/MethodLength
   # rubocop:disable Metrics/AbcSize
   def update
-    @visit = load_visit.decorate
+    @visit = load_visit
     @visit.assign_attributes(visit_params)
     @booking_response = BookingResponse.new(visit: @visit, user: current_user)
     if @booking_response.valid?
       BookingResponder.new(@booking_response, message).respond!
       flash[:notice] = t('process_thank_you', scope: [:prison, :flash])
       redirect_to visit_page(@visit)
     else
+      @visit = @visit.decorate
```

Comment:
Can you add a comment on the importance of doing this here? I still don't see how this fixes it...